libs/dh/admin/feature-user-management/src/lib/drawer/roles/tabs/content/dh-role-audit-logs.component.ts

```diff
@@ -71,37 +79,56 @@ import { DhAuditChangeCellComponent } from './dh-audit-change-cell.component';
     DhAuditChangeCellComponent,
   ],
 })
-export class DhRoleAuditLogsComponent implements OnInit, OnChanges {
-  private store = inject(DhAdminUserRoleAuditLogsDataAccessApiStore);
-
-  dataSource = new WattTableDataSource<DhRoleAuditLogEntry>();
+export class DhRoleAuditLogsComponent implements OnInit, OnDestroy, OnChanges {
+  private apollo = inject(Apollo);
+  private getUserRoleAuditLogsQuery?: QueryRef<
+    graphql.GetUserRoleAuditLogsQuery,
+    {
+      id: string;
+    }
+  >;
+  private subscription?: Subscription;
 
-  auditLogs$ = this.store.auditLogs$;
-  auditLogCount$ = this.store.auditLogCount$;
+  dataSource = new WattTableDataSource<UserRoleAuditLog>();
 
-  isLoading$ = this.store.isLoading$;
-  hasGeneralError$ = this.store.hasGeneralError$;
+  isLoading = false;
+  error?: ApolloError;
 
-  columns: WattTableColumnDef<DhRoleAuditLogEntry> = {
+  columns: WattTableColumnDef<UserRoleAuditLog> = {
     timestamp: { accessor: 'timestamp' },
-    entry: { accessor: 'entry', sort: false },
+    entry: { accessor: null },
   };
 
-  @Input() role: MarketParticipantUserRoleWithPermissionsDto | null = null;
+  @Input({ required: true }) role!: MarketParticipantUserRoleWithPermissionsDto;
 
   ngOnInit(): void {
-    this.store.auditLogs$.pipe(takeUntil(this.store.destroy$)).subscribe((logs) => {
-      this.dataSource.data = logs;
+    this.getUserRoleAuditLogsQuery = this.apollo.watchQuery({
+      useInitialLoading: true,
+      notifyOnNetworkStatusChange: true,
+      query: graphql.GetUserRoleAuditLogsDocument,
+      variables: { id: this.role.id },
     });
-  }
 
-  ngOnChanges(): void {
-    this.reloadAuditLogs();
+    this.subscription = this.getUserRoleAuditLogsQuery.valueChanges.subscribe({
+      next: (result) => {
+        this.isLoading = result.loading;
+        this.error = result.error;
+        this.dataSource.data = result.data?.userRoleAuditLogs ?? [];
+      },
+      error: (error) => {
+        this.error = error;
+      },
+    });
   }
 
-  reloadAuditLogs(): void {
-    if (this.role) {
-      this.store.getAuditLogs(this.role.id);
+  ngOnChanges(changes: SimpleChanges): void {
+    if (changes.role.currentValue) {
+      const id = changes.role.currentValue.id;
+      this.getUserRoleAuditLogsQuery?.refetch({ id });
```

Comment:
@defectiveAi I pretty sure @dzhavat and I found some issues with caching and refetch. Can you remember what that was @dzhavat ?

Reply:
Don't remember exactly 😞 

But I think the issue in our case was related to using partial data loading, which is not set up here (not needed).

The only "issue" I see is potentially double fetching of data, because a refetch is sent during `NgOnChange` and then a subscription is created during `ngOnInit`. There's only one way to find out :)